src/term_rst_post/ansimotd.py

```diff
@@ -0,0 +1,157 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this code.  If not, see <http://www.gnu.org/licenses/>.
+#
+##
+"""
+Formatting and management of MOTD text files with ANSI escape codes.
+
+@author: Alex Domingo (Vrije Universiteit Brussel)
+"""
+
+import logging
+import os
+import re
+
+from textwrap import indent, wrap
+
+from term_rst_post.exit import error_exit
+
+logger = logging.getLogger()
+
+
+def accomodate_motd(motd_dir, body_file, head_file=None, foot_file=None, foot_link=None, wrap_width=70):
+    """
+    Transform MOTD text file by: wrapping, indenting and adding header/footer
+    - motd_dir: (string) path to dir with MOTD messages
+    - body_file: (file) text file with MOTD body, it will be updated
+    - head_file: (file) text file that will be prepended to body
+    - foot_file: (file) text file that will be appended to body
+    - foot_link: (string) URL for an extra link after the body
+    - wrap_width: (int) max column width of the text file
+    """
+    motd_text = list()
+    motd_parts = [part for part in [head_file, body_file, foot_file] if part]
+    logger.debug("Parts included in the MOTD: {}".format(', '.join([part.name for part in motd_parts])))
+
+    # Prepend/Append sections to MOTD
+    for part in motd_parts:
+        try:
+            part_text = part.read()
+        except IOError as err:
+            error_exit(f"Failed to read text file: '{part.name}'", err)
+        else:
+            motd_text.append(part_text)
+            logger.info(f"Added contents of '{part.name}' to MOTD text file")
+
+    # Inject extra link between body and footer
+    if foot_link:
+        part_link = f"\nMore information in {foot_link}\n"
+        if foot_file:
+            motd_text.insert(-1, part_link)
+        else:
+            motd_text.append(part_link)
+
+    motd_text = ''.join(motd_text)
+
+    # Format MOTD text file
+    try:
+        with open(body_file.name, 'w') as motd_file:
+            motd_lines = motd_text.splitlines()
+            # Wrap text to 70 characters respecting spacings and ANSI escape codes
+            for n, line in enumerate(motd_lines):
+                motd_lines[n : n + 1] = wrap_ansicode(line, wrap_width, 5)
+            # Indent with two spaces
+            motd_lines = [indent(line, '  ') for line in motd_lines]
+            motd_file.write('\n'.join(motd_lines) + '\n')
+    except IOError as err:
+        error_exit(f"Failed to open ANSI text file to update its contents: '{body_file.name}'", err)
+    else:
+        logger.info(f"Reformatted contents of ANSI text file '{body_file.name}' to {wrap_width} characters")
+        motd_file.close()
+
+    return motd_file
+
+
+def enable_motd(motd_cfg, motd_file):
+    """
+    Enable MOTD message
+    - motd_cfg: (string) path to config file in MOTD directory
```

Comment:
is the motd_cfg still relevant with this?

Reply:
very good point, given how the Azure pipeline works this is indeed unnecessary. I removed it in https://github.com/sisc-hpc/term-rst-post/pull/1/commits/6832337c17938e38276f71f66dd8500e6c10f91c. Now `news2motd` generates a file with a known name, which can be controlled with `--motd-path`.